src/pip/_internal/utils/misc.py

```diff
@@ -65,9 +65,9 @@
 except ImportError:
     # typing's cast() isn't supported in code comments, so we need to
     # define a dummy, no-op version.
-    def cast(typ, val):
+    def cast(typ, val):  # type: ignore
```

Comment:
I would add a code comment saying it's okay to ignore the type info here since if we're in the except block, we know we're not in the process of checking types (which isn't true for all such `ImportError` except blocks).

Reply:
TBH, I want to just change this up to get rid of the "from typing import ..." that may execute at runtime and add an else clause to the if mypy check above.

I'd prefer to do that in a follow up though.

Reply:
I'll add this change to this PR, since it's pretty trivial.